pkg/broker/wait_for_children.go

```diff
@@ -0,0 +1,103 @@
+package broker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Azure/open-service-broker-azure/pkg/async/model"
+	log "github.com/Sirupsen/logrus"
+)
+
+func (b *broker) doWaitForChildrenStep(
+	ctx context.Context,
+	args map[string]string,
+) error {
+
+	_, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
This looks like a mistake. This will cancel the context when this function returns, but this context isn't being passed to any functions. (It can't be, because we've used `_` to refuse assigning it to any variable. If `ctx`, is passed to any functions below [I haven't looked yet], that is the `ctx` _we were passed_ and _that_ `ctx` won't be canceled when this function returns.